module/zfs/zpl_super.c

```diff
@@ -248,25 +248,67 @@ zpl_fill_super(struct super_block *sb, void *data, int silent)
 	return (error);
 }
 
-#ifdef HAVE_MOUNT_NODEV
+static int
+zpl_test_super(struct super_block *s, void *data)
+{
+	zfsvfs_t *zfsvfs = s->s_fs_info;
+	objset_t *os = data;
+
+	return (os == zfsvfs->z_os);
+}
+
+static struct super_block *
+zpl_mount_impl(struct file_system_type *fs_type, int flags,
+    zfs_mnt_t zm)
```

Comment:
nit: this fits on the previous line, and you should pass this as a `zfs_mnt_t *`.